### ISSUE TITLE
HTTP client doesn't follow redirect

### ISSUE THREAD

**Issue by jonesbusy** [jenkinsci/plugin-modernizer-tool#255]
### Jenkins and plugins versions report

```
Fetching data from: https://updates.jenkins.io/current/plugin-versions.json 
io.jenkins.tools.pluginmodernizer.core.model.ModernizerException: Failed to get JSON data: 307
        at io.jenkins.tools.pluginmodernizer.core.utils.JsonUtils.fromUrl(JsonUtils.java:96)
```

### What Operating System are you using (both controller, and any agents involved in the problem)?

All

### Reproduction steps

Just start with a fresh cache

### Expected Results

It works

### Actual Results

It doesn't work and doesn't fetch data from update center

### Are you interested in contributing a fix?

Yes PR incoming

**Comment by jonesbusy** [comment]
On my side the update center redirect me to `https://westeurope.cloudflare.jenkins.io/current/update-center.actual.json`

**Comment by lemeurherve** [comment]
Cc @dduportal

Le jeu. 12 sept. 2024, 05:43, Valentin Delaye ***@***.***> a
écrit :

**Comment by dduportal** [comment]
Hi @jonesbusy ! This unexpected behavior is definitively a consequence of the Update Center Brownout currently in progress which introduces redirections to the mirror serving the expected data (close to your location).

Happy to see the issue has been resolved!

Note: we also had the problem (described in https://github.com/jenkins-infra/helpdesk/issues/2649#issuecomment-2345352162) where the new Update Center responded with its own JSON API content when the client has an Accept Header set to "application/json" (which is .. expected for https://updates.jenkins.io/current/plugin-versions.json 😅) instead of being redirected to the expected file.
We fixed this behavior **after** your fixed your problem here. Mentioning just in case.
